### ISSUE TITLE
pylint ignores visit_Constant with python >3.8

### ISSUE THREAD

**Issue by Cielquan** [PyCQA/pylint#4054]
### Steps to reproduce
1. create a test `py` file with this content:
```python
# pylint: disable=C0114,C0115,C0116
import ast


class Visitor(ast.NodeVisitor):

    def visit_Constant(self, node):
        return node.value

    def visit_Num(self, node):
        return node.n
```
2. run `pylint` via python 3.7 or older
3. see expected output:
```
test.py:7:4: C0103: Method name "visit_Constant" doesn't conform to snake_case naming style (invalid-name)
test.py:7:4: R0201: Method could be a function (no-self-use)
test.py:10:4: C0103: Method name "visit_Num" doesn't conform to snake_case naming style (invalid-name)
test.py:10:4: R0201: Method could be a function (no-self-use)
```
4. run `pylint` via python 3.8 or newer
5. see unexpected output:
```
test.py:10:4: C0103: Method name "visit_Num" doesn't conform to snake_case naming style (invalid-name)
test.py:10:4: R0201: Method could be a function (no-self-use)
```

### Current behavior
> Deprecated since version 3.8: Methods visit_Num(), visit_Str(), visit_Bytes(), visit_NameConstant() and visit_Ellipsis() are deprecated now and will not be called in future Python versions. Add the visit_Constant() method to handle all constant nodes.

Source: https://docs.python.org/3.8/library/ast.html#ast.NodeVisitor

The `ast.NodeVisitor.visit_Constant()` method was added in python 3.8. Strangely the method is completely ignored by `pylint` when run via python3.8 or newer (where it was added). Only with python versions prior python3.8 the method gets recognized.

### Expected behavior
Both methods (`visit_Constant` and `visit_Num`) should be recognized regardless of minor python version.

### pylint --version output
Tested with the following versions:

pylint 2.6.0
astroid 2.4.2
Python 3.9.1 (default, Dec  8 2020, 02:26:20) 
[GCC 9.3.0]

pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0]

pylint 2.6.0
astroid 2.4.2
Python 3.7.9 (default, Aug 18 2020, 02:07:21) 
[GCC 9.3.0]

pylint 2.6.0
astroid 2.4.2
Python 3.6.12 (default, Aug 17 2020, 23:45:20) 
[GCC 9.3.0]
